packages/@aws-cdk/aws-ec2/lib/private/ebs-util.ts

```diff
@@ -34,9 +35,19 @@ function synthesizeBlockDeviceMappings<RT, NDT>(construct: Construct, blockDevic
       } else if (volumeType !== EbsDeviceVolumeType.IO1) {
         Annotations.of(construct).addWarning('iops will be ignored without volumeType: EbsDeviceVolumeType.IO1');
       }
+
+      cfnEbs = {
+        ...ebs,
```

Comment:
Ah hah.

The problem is that this object still contains the `kmsKey` object, which is why the tests are failing with this error:

```
@aws-cdk/aws-ec2:     Resolution error: Resolution error: Unable to resolve object tree with circular reference. Path: /Resources/${Token[Default.Instance.Resource.LogicalID.983]}/Properties/blockDeviceMappings/0/ebs/kmsKey/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host/node/host..
```

It seems like the fields were previously INCIDENTALLY identical to the `CfnEbs` object, but no longer. You need to copy the fields out individually (and transform them as necessary).